cmd/proxy/actions/basicauth_test.go

```diff
@@ -1,57 +1,80 @@
 package actions
 
 import (
+	"bytes"
+	"context"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 
-	"github.com/stretchr/testify/require"
+	"github.com/gomods/athens/pkg/log"
+	"github.com/sirupsen/logrus"
 )
 
-type checkAuthTest struct {
-	name     string
-	user     string
-	pass     string
-	httpUser string
-	httpPass string
-	want     bool
-}
-
-var checkAuthTests = [...]checkAuthTest{
+var basicAuthTests = [...]struct {
+	name   string
+	user   string
+	pass   string
+	path   string
+	logs   string
+	status int
+}{
 	{
-		name:     "success",
-		user:     "athens",
-		pass:     "1234",
-		httpUser: "athens",
-		httpPass: "1234",
-		want:     true,
+		name:   "happy_path",
+		user:   "correctUser",
+		pass:   "correctPass",
+		path:   "/",
+		logs:   "",
+		status: 404,
 	},
 	{
-		name:     "username incorrect",
-		user:     "athens",
-		pass:     "1234",
-		httpUser: "not athens",
-		httpPass: "1234",
-		want:     false,
+		name:   "incorrect_username",
+		user:   "wrongUser",
+		pass:   "correctPass",
+		path:   "/",
+		logs:   "",
+		status: 401,
 	},
 	{
-		name:     "password incorrect",
-		user:     "athens",
-		pass:     "1234",
-		httpUser: "athens",
-		httpPass: "not 1234",
-		want:     false,
+		name:   "incorrect_password",
+		user:   "correctUser",
+		pass:   "wrongPassword",
+		path:   "/",
+		logs:   "",
+		status: 401,
+	},
+	{
+		name:   "log_on_healthz",
+		user:   "wrongUser",
+		pass:   "wrongPassword",
+		path:   "/healthz",
+		logs:   healthWarning,
+		status: 401,
```

Comment:
can we name it expectedStatus or something similar which tells that we are checking against it and not actually passing in